### PR TITLE
Project

### PR DESCRIPTION
**Pull request by szczepanskikrs**
### Description

Created basic project outline and structure. Added maven support and basic dependencies. 

### List general changes for this pull request:
* Added .gitignore
* Added Licence
* Added Maven support
* Added Travis CI support and travis.yml.
* Added Basic project structure
* Added Dependencies
* Added new modules
* Added pullrequest and issue templates.

### List of modules this pull request affects:
* scrapper
* database
* application

### CODE REVIEW

File: application/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    </description>
+
+    <parent>
+        <artifactId>robot</artifactId>
+        <groupId>com.robot</groupId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
+        <!--For testing-->
```
Comment (korotkevics) [inline application/pom.xml]: Maybe just "Testing"?

File: database/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    </description>
+
+    <parent>
+        <artifactId>robot</artifactId>
+        <groupId>com.robot</groupId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
+        <!--For testing-->
```
Comment (korotkevics) [inline database/pom.xml]: Maybe just "Testing"?

File: pom.xml [hunk 3]
```diff
@@ -3,10 +3,119 @@
...
     <version>1.0-SNAPSHOT</version>
+    <modules>
+        <module>scrapper</module>
+        <module>application</module>
+        <module>database</module>
+    </modules>
+
+    <properties>
+        <!--Encoding-->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (korotkevics) [inline pom.xml]: <project.build.targetEncoding>UTF-8</project.build.targetEncoding>
<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>

File: pom.xml [hunk 4]
```diff
@@ -3,10 +3,119 @@
...
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <catalina.base>/var/lib/tomcat8</catalina.base>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <!--so maven would not fail on missing web.xml-->
+        <failOnMissingWebXml>false</failOnMissingWebXml>
+        <!--Hibernate & DB-->
+        <postgres.version>9.1-901-1.jdbc4</postgres.version>
+        <h2.vesrion>1.4.196</h2.vesrion>
+        <hibernate.version>5.2.13.Final</hibernate.version>
+        <!--Spring & other stuff-->
```
Comment (korotkevics) [inline pom.xml]: other stuff -> related

File: pom.xml [hunk 5]
```diff
@@ -3,10 +3,119 @@
...
+    <properties>
+        <!--Encoding-->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <!--Compiling-->
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <catalina.base>/var/lib/tomcat8</catalina.base>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <!--so maven would not fail on missing web.xml-->
+        <failOnMissingWebXml>false</failOnMissingWebXml>
+        <!--Hibernate & DB-->
```
Comment (korotkevics) [inline pom.xml]: Hibernate & DB -> DB

File: scrapper/pom.xml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>scrapper</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>war</packaging>
+    <name>scrapper</name>
+
+    <description>
+        Html parser and scrapper.
```
Comment (korotkevics) [inline scrapper/pom.xml]: HTML

File: scrapper/pom.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    </description>
+
+    <parent>
+        <artifactId>robot</artifactId>
+        <groupId>com.robot</groupId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
```
Comment (korotkevics) [inline scrapper/pom.xml]: scope is inherited, remove

Comment (szczepanskikrs) [reply scrapper/pom.xml]: :+1:

File: pom.xml [hunk 8]
```diff
@@ -3,10 +3,119 @@
...
+        <testng.version>6.14.2</testng.version>
+        <assertj.version>3.9.0</assertj.version>
+        <mockito.version>2.15.0</mockito.version>
+
 
+    </properties>
 
+    <dependencyManagement>
+        <dependencies>
+            <!--Spring related dependencies.-->
```
Comment (korotkevics) [inline pom.xml]: Spring related would suffice

File: pom.xml [hunk 9]
```diff
@@ -3,10 +3,119 @@
...
+                <groupId>org.springframework.security</groupId>
+                <artifactId>spring-security-web</artifactId>
+                <version>${spring.framework.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.springframework.security</groupId>
+                <artifactId>spring-security-config</artifactId>
+                <version>${spring.framework.version}</version>
+            </dependency>
+            <!--Databases and hibernate related dependencies.-->
```
Comment (korotkevics) [inline pom.xml]: remove "dependencies"

**Comment by korotkevics** [comment]
it does not contain .gitignore

**Pull request closed without merging**